src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/IntegrationTests/AdminServiceTests.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using DevAdventCalendarCompetition.Repository;
+using DevAdventCalendarCompetition.Repository.Context;
+using DevAdventCalendarCompetition.Repository.Models;
+using DevAdventCalendarCompetition.Services;
+using DevAdventCalendarCompetition.Services.Models;
+using DevAdventCalendarCompetition.Services.Profiles;
+using FluentAssertions;
+using Xunit;
+
+namespace DevAdventCalendarCompetition.Tests.IntegrationTests
+{
+    public class AdminServiceTests : IntegrationTestBase
+    {
+        public AdminServiceTests()
+            : base()
+        {
+        }
+
+        [Fact]
+        public void Gets_all_tests()
+        {
+            var testList = GetTestList();
+            using (var context = new ApplicationDbContext(this.ContextOptions))
```

Comment:
why do you create dbContext twice?

Reply:
The reason for this is that the setting up of the database leaves tracked data inside the application’s DbContext. The state of the application’s DbContext isn’t the same as what the method under test would encounter in web application. Sometimes it doesn’t matter, but sometimes it does—which can lead to subtle bugs being missed.

Reply:
ok, for test purposes it is acceptable - we don't want to use cache ;) in other project we just use transactions ;p